python/pyspark/ml/classification.py

```diff
@@ -2274,21 +2276,21 @@ def setSolver(self, value):
         return self._set(solver=value)
 
 
-class MultilayerPerceptronClassificationModel(JavaProbabilisticClassificationModel, JavaMLWritable,
+class MultilayerPerceptronClassificationModel(JavaProbabilisticClassificationModel,
+                                              _MultilayerPerceptronParams, JavaMLWritable,
                                               JavaMLReadable):
     """
     Model fitted by MultilayerPerceptronClassifier.
 
     .. versionadded:: 1.6.0
     """
 
-    @property
-    @since("1.6.0")
-    def layers(self):
+    @since("3.0.0")
+    def setLayers(self, value):
```

Comment:
Do we need to add this setter? Do we allow to change layers of a model?

Reply:
I guess this method is for old version. Since the new version return layers via $(layers), so we need to manually set this param in old version.

Reply:
When we need to manually set layers param? I think it is fixed after training?

And we don't have other setters like setMaxIter.

Reply:
We manually set layers in loading the old model in scala. I think we normally keep the same methods between scala and python, so I put this in python too.

Reply:
Agree. I will use  ```model.set("layers", layers)``` in load. Will remove ```setLayers```.

Reply:
I opened the jira ticket and want to thank everyone for their work and comments. @viirya - I'm not sure I understood your question about setters, however, from a user perspective (me) the important thing is to expose what parameters were used to train the model. After running a cross validation, there is no other way of accessing parameters to analyze the results of the grid search. There is also no built in way to log the parameters during each grid search attempt.

Here you can see an example of how that is useful to narrow down the hyperparameter space based on which hyperparameters generally give good results.

https://github.com/CyborgDroid/spark_mlflow/blob/develop/README.md#mlflow-to-analyze-best-hyperparameters-for-further-optimization

Reply:
You can get the parameters used in training by getters. Getters are exposed.

Reply:
The purpose of the extractParamMap function is to expose all parameters used to train the model. Having to execute all getters is not a good approach or solution.

Reply:
Why extractParamMap does not work without setters?